npm_and_yarn/helpers/lib/npm/conflicting-dependency-parser.js

```diff
@@ -43,4 +40,31 @@ async function findConflictingDependencies(directory, depName, targetVersion) {
   });
 }
 
+function buildExplanation(node, directEdge, topLevelEdge) {
+  if (directEdge.from === topLevelEdge.to) {
+    // The nodes parent is top-level
+    return `${directEdge.from.name}@${directEdge.from.version} requires ${directEdge.to.name}@${directEdge.spec}`;
+  } else if (topLevelEdge.to.edgesOut.has(directEdge.from.name)) {
+    // The nodes parent is a direct dependency of the top-level dependency
+    return `${topLevelEdge.to.name}@${topLevelEdge.to.version} requires ${directEdge.to.name}@${directEdge.spec} via ${directEdge.from.name}@${directEdge.from.version}`;
+  } else {
+    // The nodes parent is a transitive dependency of the top-level dependency
+    return `${topLevelEdge.to.name}@${topLevelEdge.to.version} requires ${directEdge.to.name}@${directEdge.spec} via a transitive dependency on ${directEdge.from.name}@${directEdge.from.version}`;
```

Comment:
Maybe cut up the line a bit?
```suggestion
    return `${topLevelEdge.to.name}@${topLevelEdge.to.version} requires ${directEdge.to.name}@${directEdge.spec} ` +
               `via a transitive dependency on ${directEdge.from.name}@${directEdge.from.version}`;
```

Reply:
Yeah good call, prettier doesn't like that specific suggestion, but I've pushed an alternative 👍